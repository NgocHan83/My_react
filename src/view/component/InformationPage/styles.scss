.wrapper-form {
  background: #eaf1cf;
  width: 100%;
  height: 100%;
  background-size: cover;
  position: absolute;
  .form-info {
    width: 400px;
    border: 1px solid #000;
    border-radius: 5px;
    position: absolute;
    background-color: #228B22;
    left: 50%;
    -webkit-transform: translate(-50%, 0);
    top: calc(50vh - 400px);
    padding: 15px;
    box-shadow: 0 6px 10px 0 rgba(0, 0, 0, 0.8);
    label {
      color: #fff;
    }
    
    .female {
     padding-left: 20px;
    }
    
    input[type='radio'] {
      width: 18px;
      height: 18px;
    }

    .wrapper-btn {
      text-align: center;
      input {
        text-transform: lowercase;
      }
      fieldset input:first-letter {
        text-transform:capitalize !important
      }
      
    }
    .btn-primary {
      width: 50%;
      height: 45px;
      background-color: #FF8C00;
      border-color: #FF8C00;
      border-radius: 22px;
      font-size: 18px;
      box-shadow: 0 6px 10px 0 rgba(0, 0, 0, 0.2);
    }
    .btn-primary:focus {
      background-color: #FF8C00;
      border-color: #FF8C00;
      box-shadow: 0 0 0 0;
     }
    
    .btn-primary:hover {
      background-color: #FF8C00;
      border-color: #FF8C00;
      box-shadow: none;
      -webkit-box-shadow:none;
     }
    
    .btn-primary:not(:disabled):not(.disabled):active,
    .show>.btn-primary.dropdown-toggle { 
      background-color: #FF8C00;
      border-color: #FF8C00;
      background-image: linear-gradient(241deg, #e95707, #f58220);
     }
    
    .btn-primary:not(:disabled):not(.disabled):active:focus,
    .show>.btn-primary.dropdown-toggle {
      box-shadow: 0 0 0 0 #FF8C00;
     }
    
    .btn-primary:disabled {
      background-color: #FF8C00;
      border-color: #FF8C00;
     } 
  }  

  // .capitalize {
  //   text-transform: lowercase;
  //   display: inline-block;
  // }
    
}

