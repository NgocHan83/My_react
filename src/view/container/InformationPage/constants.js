export const INPUT_DATA_INF_START = 'INPUT_DATA_INF_START';