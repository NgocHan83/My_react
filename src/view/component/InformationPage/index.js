import React, { Component } from 'react';
import { Field } from 'redux-form';
import '../../../../node_modules/bootstrap/dist/css/bootstrap.min.css';

import './styles.scss';

class InformationPage extends Component {
 
  render() {
    const { onSubmit, pristine } = this.props;

    return (
      <div className='wrapper-form'>
        <form className='form-info' onSubmit={onSubmit}>
          <div className="form-group">
            <label htmlFor='firstName'>First Name</label>
            <div>
              <Field
                className='form-control'
                name='firstName'
                type='text'
                component='input'
                placeholder='First Name' />
            </div>
          </div>
          <div className='form-group'>
            <label htmlFor='lastName'>Last Name</label>
            <div>
              <Field
                className='form-control'
                name='lastName'
                type='text'
                component='input'
                placeholder='Last Name' />
            </div>
          </div>
          <div className='form-group'>
            <label htmlFor='email' >Email</label>
            <div>
              <Field
                className='form-control'
                name='email'
                type='email'
                component='input'
                placeholder='Email' />
            </div>
          </div>
          <div className='form-group'>
            <label htmlFor='gender'>Gender</label>
            <div>
              <label className='male'>
                <Field
                  name='gender'
                  type='radio'
                  component='input'
                  value='male' /> {' '} Male
              </label>
              <label className='female'>
                <Field
                  name='gender'
                  type='radio'
                  component='input'
                  value='female' /> {' '} Female
              </label>
            </div>
          </div>
          <div className='form-group wrapper-btn'>
            <button className='btn btn-primary' disabled={pristine}>Submit</button>
            <input type='text' className='dv' />
            <p className='dv'>HAN HAN</p>
          </div>
        </form>
      </div>    )
  }
}

export default InformationPage;