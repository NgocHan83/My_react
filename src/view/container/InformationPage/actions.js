import { INPUT_DATA_INF_START } from './constants';

export const inputDataInfStart = (data) => ({
  type: INPUT_DATA_INF_START,
  data
});