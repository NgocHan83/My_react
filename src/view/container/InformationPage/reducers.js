import { INPUT_DATA_INF_START } from './constants';

const initialState = {
  inputData: {
    firstName: 'Trinh',
    lastName: 'Han',
    email: 'ttngochan83@gmail.com',
    gender: 'female'
  }
};

const inputDataReducer = (state = initialState, action) => {
  const { type } = action;
  switch (type) {
    case INPUT_DATA_INF_START:
      return {
        ...state,
        inputData: {
          ...state.inputData,
          firstName: action.firstName,
          lastName: action.lastName,
          email: action.email,
          gender: action.gender
        }
      }
    default:
      return state;
  }
};

export default inputDataReducer;