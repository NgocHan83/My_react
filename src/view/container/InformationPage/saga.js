import { takeEvery, put } from 'redux-saga/effects';
import { reset } from 'redux-form';
import { INPUT_DATA_INF_START } from './constants';


function* inputDataInfWorker() {
  try {
    yield put(reset('informationForm'));
  } catch (err) {
    throw (err)
  }
}

const inputWatchers = [
  takeEvery(INPUT_DATA_INF_START, inputDataInfWorker)
];

export default inputWatchers;