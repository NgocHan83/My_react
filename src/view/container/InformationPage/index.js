import React, { Component } from 'react';
import { reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { inputDataInfStart } from './actions';
import InformationPage from '../../component/InformationPage';

class InformationContainer extends Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onSubmit(value) {
    const { inputData } = this.props;
    console.log()
    alert('Submit successed!')
    inputData(value);
  }

  render() {
    const { handleSubmit, pristine } = this.props;
    return (
      <InformationPage onSubmit={handleSubmit(this.onSubmit)} pristine={pristine} />
    )
  }
}
const formInfomation = reduxForm({
  form: 'informationForm',
  enableReinitialize: true,
  keepDirtyOnReinitialize: true
})(InformationContainer);

const mapStateToProps = ({ inputDataReducer }) => ({
  initialValues: inputDataReducer.inputData
});

const mapDispatchToProps = dispatch => ({
  inputData: (value) => dispatch(inputDataInfStart(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(formInfomation);