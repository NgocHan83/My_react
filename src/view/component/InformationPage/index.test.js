import React from 'react';
import ReactDOM from 'react-dom';
import InformationPage from './index';
import renderer from 'react-test-renderer';


describe('Information Page component', () => {
  it ('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InformationPage></InformationPage>, div); 
    ReactDOM.unmountComponentAtNode(div); 
  });

  test('has a valid snapshot', () => { 
    const component = renderer.create( <InformationPage />); 
    let tree = component.toJSON(); expect(tree).toMatchSnapshot(); 
  });

})
