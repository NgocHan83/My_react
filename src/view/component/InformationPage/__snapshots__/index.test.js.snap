// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Page component has a valid snapshot 1`] = `
<div
  className="wrapper-form"
>
  <form
    className="form-info"
    onSubmit={undefined}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="firstName"
      >
        First Name
      </label>
      <div />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="lastName"
      >
        Last Name
      </label>
      <div />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <div />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="gender"
      >
        Gender
      </label>
      <div>
        <label
          className="male"
        />
        <label
          className="female"
        />
      </div>
    </div>
    <div
      className="form-group wrapper-btn"
    >
      <button
        className="btn btn-primary"
        disabled={undefined}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
